p/src/PlumeStaking.sol

```diff
@@ -448,10 +621,12 @@ contract PlumeStaking is Initializable, AccessControlUpgradeable, UUPSUpgradeabl
         PlumeStakingStorage storage $ = _getPlumeStakingStorage();
         StakeInfo storage info = $.stakeInfo[user];
 
-        amount = info.accumulatedRewards;
+        uint256 pending = 0;
         if (info.staked > 0 && block.timestamp > info.lastUpdateTimestamp) {
-            amount += (info.staked * (block.timestamp - info.lastUpdateTimestamp) * $.perSecondRewardRate) / _BASE;
+            pending =
+                (info.staked * (block.timestamp - info.lastUpdateTimestamp) * $.rewardRates[address($.pUSD)]) / _BASE;
         }
+        amount = $.rewardAccrued[user][address($.pUSD)] + pending;
     }
 
 }
```

Comment:
I think "stake" and "parkAndStake" are redundant, due to stake not doing a complete job, it just internally bookkeep the amount from parked to staked, and users don't need to actually transfer amount in, while it still takes an external call to do the bookkeeping.

I think we should:

1. implement stake with the same logic as "parkAndStake"
2. remove parkAndStake
3. implement "unstake" the same as the current unstake, no change
4. implement "claim_rewards" the same ways as "claim" but using "claim_rewards" to be more explicit
5. "withdraw_principle" to do the same logic as "unpark"